### PR TITLE
105471 join trust changes

### PR DESCRIPTION
**Pull request by paullocknimble**
Added ChangesToTrust and ChangesToTrustExplained to the JoinTrust domain object

Also as part of this change I have brought in automapper to relieve the burden of manual mapping
The mappings, trust only to start with, are currently set up in the webapi which is not ideal as I wanted to push this down to the responsibility of the service layer but because of how the projects have been separated I wouldn't have access to the types to form the mapping config. One for the future. One other thing to note is the IMapper interface is passed into the original manual mapper for now due to still manually mapping all other objects.

Unit tests have been put around the mappings, as it stands one object goes through 3 layers, so three maps, domain to state, state to domain and domain to service model.

Importantly all mappings are verified in units tests with this line mapperConfig.AssertConfigurationIsValid();

so shouldn't ever be deployed in a state where the mappings are broken or invalid.

### CODE REVIEW

**Review by jarednimble — APPROVED** [review]

File: Dfe.Academies.Academisation.WebApi.UnitTest/AutoPopulatedMoqPropertiesCustomization.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	public class AutoPopulatedMoqPropertiesCustomization : ICustomization
+	{
+		public void Customize(IFixture fixture)
+		{
+			fixture.Customizations.Add(
+				new PropertiesPostprocessor(
+					new MockPostprocessor(
+						new MethodInvoker(
+							new MockConstructorQuery()))));
+			fixture.ResidueCollectors.Add(new MockRelay());
```
Comment (MarkRobinsonDfe) [inline Dfe.Academies.Academisation.WebApi.UnitTest/AutoPopulatedMoqPropertiesCustomization.cs:17]: I think I have a residue collector in my car. is this the same thing?

File: Dfe.Academies.Academisation.IService/ServiceModels/Application/ApplicationJoinTustServiceModel.cs [hunk 2]
```diff
@@ -6,5 +6,5 @@
 
 namespace Dfe.Academies.Academisation.IService.ServiceModels.Application
 {
-	public record ApplicationJoinTustServiceModel(int id,  string trustName, int UkPrn);
+	public record ApplicationJoinTustServiceModel(int Id,  int UKPRN, string TrustName, bool? ChangesToTrust, string? ChangesToTrustExplained);
```
[the remark this answers is not in the document]
Comment (paullocknimble) [reply Dfe.Academies.Academisation.IService/ServiceModels/Application/ApplicationJoinTustServiceModel.cs:9]: ha, yes I will change this

[the remark this answers is not in the document]
Comment (MarkRobinsonDfe) [reply Dfe.Academies.Academisation.IService/ServiceModels/Application/ApplicationJoinTustServiceModel.cs:9]: a lot worse, yes

File: Dfe.Academies.Academisation.WebApi.UnitTest/AutoPopulatedMoqPropertiesCustomization.cs (hunk 1 above)
Comment (MarkRobinsonDfe) [reply Dfe.Academies.Academisation.WebApi.UnitTest/AutoPopulatedMoqPropertiesCustomization.cs:17]: okey dokey

**Pull request merged**